HABApp/openhab/http_connection.py

```diff
@@ -320,6 +329,44 @@ async def async_get_item(self, item_name: str) -> dict:
         else:
             return await ret.json(encoding='utf-8')
 
+    async def async_thing_remove_link(self, thing_uid: str, channel: str, item_name: str) -> bool:
+        if self.config.general.listen_only:
+            return False
+
+        channel_uid = f"{thing_uid}:{channel}"
+        url = self.__get_openhab_url('rest/links/{item:s}/{channel:s}', item=item_name, channel=channel_uid)
+        
+        fut = self.__session.delete(url, json={})
+        
+        ret = await self._check_http_response(fut)
+        return ret.status == 200
+
+    async def async_thing_link_exists(self, thing_uid: str, channel: str, item_name: str) -> bool:        
+        channel_uid = f"{thing_uid}:{channel}"
+        url = self.__get_openhab_url('rest/links/{item:s}/{channel:s}', item=item_name, channel=channel_uid)
+        
+        fut = self.__session.get(url,json={})
+
+        ret = await self._check_http_response(fut, accept_404=True)
+        return ret.status == 200
+
+    async def async_thing_add_link(self, thing_uid: str, channel: str, item_name: str) -> bool:
+        if self.config.general.listen_only:
+            return False
+
+        # check for item existence first, otherwhise OpenHAB creates a new item when we add a link with an unknown itemname
+        exists = await self.async_item_exists(item_name)        
+        if not exists:
+            raise ItemNotFoundException(f'Item "{item_name}" does not exist')
+        
+        channel_uid = f"{thing_uid}:{channel}"
+        url = self.__get_openhab_url('rest/links/{item:s}/{channel:s}', item=item_name, channel=channel_uid)
+        
+        fut = self.__session.put(url, json={})
+
+        ret = await self._check_http_response(fut)
+        return ret.status == 200
```

Comment:
```suggestion
    async def async_thing_add_link(self, thing_uid: str, channel: str, item_name: str, config: typing.Optional[ItemChannelLinkDefinition]) -> bool:
        if self.config.general.listen_only:
            return False

        # check for item existence first, otherwhise OpenHAB creates a new item when we add a link with an unknown itemname
        exists = await self.async_item_exists(item_name)        
        if not exists:
            raise ItemNotFoundException(f'Item "{item_name}" does not exist')
        
        url = self.__get_openhab_url('rest/links/{item:s}/{thing_uid:s}:{channel:s}', item=item_name, thing_uid=thing_uid, channel=channel_uid)
        
        fut = self.__session.put(url, json=o.dict(by_alias=True) if isinstance(config, ItemChannelLinkDefinition) else None)

        ret = await self._check_http_response(fut)
        return ret.status == 200
```

Reply:
This would allow passing in ItemChannelLinkDefinition objects

Reply:
When using the ItemChannelLinkDefinition we could also use it as a parameter directly and remove item_name, thing_name and channel because item_name and channel_uid are properties of the definition.

To me the usage of the definitions is incosistent currently. Some methods in http_connection return the parsed item definition directly, others a dict which is parsed into the definition in oh_interface. One could also use the OpenhabItemDefinition as a parameter in async_create_item instead of the long parameter list.

Is it possible to add methods to such a definition class?

Reply:
I found a [good description of the configuration](https://github.com/eclipse/smarthome/blob/11764d8ae824cadc95de5dd228f9f077f4859dcc/docs/documentation/development/bindings/profiles.md).
```
Number <itemName> { channel="<bindingID>:<thing-typeID>:<thingName>:<channelName>"[profile="offset", offset="<value>"]}
```
the ```[profile="offset", offset="<value>"]``` describes the configuration meaning there is a key for the profile and as many other keys as needed to configure the profile.
I'm not sure if it is necessary to model this as class or just have the dict<str,Any> available if needed.

Reply:
> When using the ItemChannelLinkDefinition we could also use it as a parameter directly and remove item_name, thing_name and channel because item_name and channel_uid are properties of the definition.

Sounds good

> Is it possible to add methods to such a definition class?

Sure, but let's finish one PR at a time. I'm excited that someone is collaborating but unfortunately I have not very much experience using the review tools from github.